lib/kube/kubeconfig/kubeconfig.go

```diff
@@ -210,9 +219,36 @@ func UpdateConfig(path string, v Values, storeAllCAs bool, fs ConfigFS) error {
 			if len(envVars) > 0 {
 				authInfo.Exec.Env = envVars
 			}
-			config.AuthInfos[authName] = authInfo
+			config.AuthInfos[contextName] = authInfo
+
+			setStringExtensionInAuthInfo(config.AuthInfos[contextName], extTeleClusterName, v.TeleportClusterName)
+			setStringExtensionInAuthInfo(config.AuthInfos[contextName], extKubeClusterName, c)
+
+			config.Clusters[contextName] = &clientcmdapi.Cluster{
+				Server:                   v.ClusterAddr,
+				CertificateAuthorityData: cas,
+				TLSServerName:            v.TLSServerName,
+			}
 
-			setContext(config.Contexts, contextName, clusterName, authName, c, v.Namespace)
+			setStringExtensionInCluster(config.Clusters[contextName], extTeleClusterName, v.TeleportClusterName)
+			setStringExtensionInCluster(config.Clusters[contextName], extKubeClusterName, c)
+
+			// XXX(espadolini): this assumes (as just about everything in
+			// tsh does) that the profile name is exactly the same as the
+			// hostname of the Proxy web address, and it would be more correct
+			// to update every callsite to also pass the profile name
+			profileName, err := utils.Host(v.ProxyAddr)
```

Comment:
nit: profileName shouldn't be empty in this case. should we just error out?

Reply:
`tsh login --format kubernetes --out foo` and `tctl auth sign --format kubernetes` don't pass a proxy addr because the `identityfile` interface doesn't support that, but the output file is always a custom path and it never needs explicit cleanup (even if it's merged with an existing kubeconfig) so I think it's ok. Maybe we should error out if there's a nonempty `ProxyAddr` but there's an error from `utils.Host` tho.